set.go

```diff
@@ -140,3 +140,49 @@ func SetInt(node *goyaml.Node, path Path, value int64) (valueNode *goyaml.Node,
 	})
 	return
 }
+
+// SetScalar sets a scalar value at the given path within the given node.
+//
+// node is the node to set the value within.
+//
+// path is the path within the node to set the value at.
+//
+// value is the scalar value to set, eg. "hello".
+//
+// Returns a bool showing whether or not the value set updated an existing value (ie. value was changed).
+func SetScalar(node *goyaml.Node, path Path, value any) (changed bool, err error) {
+	n, err := plumbing.MakeScalar(value)
+	if err == nil {
+		_, changed, err = set(node, path, nil, func() *goyaml.Node {
```

Comment:
Bugfix. Add test cases for this: https://github.com/jbrekelmans/go-edityaml/pull/5/files#r1577151228
```suggestion
		_, changed, err = set(node, path, value, func() *goyaml.Node {
```